### PR TITLE
Bring json-editor changes into digest lifecycle

### LINKED ISSUE #35: Schema save button doesn't disable correctly
See TODO in schema/edit-controller.js

### PR DESCRIPTION
**Pull request by moradology**
closes #35

### CODE REVIEW

**Comment by moradology** [comment]
@kshepard @ddohler
After I tracked this down, I asked Andrew for some guidance about how best to deal with $apply - this is where we landed.

File: schema_editor/app/scripts/json-editor/json-editor-directive.js [hunk 1]
```diff
@@ -18,7 +18,8 @@
 
     var defaults = {};
 
-    function JsonEditor() {
+    /* ngInject */
+    function JsonEditor($timeout) {
```
Comment (kshepard) [inline schema_editor/app/scripts/json-editor/json-editor-directive.js]: Is the $timeout needed?

**Comment by kshepard** [comment]
Interesting. Nice find!

**Comment by kshepard** [comment]
http://nathanleclaire.com/blog/2014/01/31/banging-your-head-against-an-angularjs-issue-try-this/

**Comment by moradology** [comment]
"I care, I share, I'm Nathan LeClaire."

**Comment by CloudNiner** [comment]
+1

**Pull request merged**